python/lsst/analysis/drp/calcFunctors.py

```diff
@@ -0,0 +1,39 @@
+from lsst.analysis.drp.configStructField import Config, Field
+import numpy as np
+
+
```

Comment:
Tim, I have been helping Sophie with some of this, and wanted to chime in here. This is not going to be a complete response, because I would like to take a bit of time to make a proposal for the two of you. But, I wanted to take minute to acknowledge your comments, and let you know that it is in preparation.

I certainly agree that if possible it would be good to bring these objects together. From what I have looked into thus far this would be somewhat an amalgamation of both, but really this depends of course on details, so I dont want to say anything more until I have more of a defendable proposal.

One thing that did catch my eye with this is the name of the existing `Functors`. We absolutely must rename that object. A functor is a generic idea, and can be found in various parts of the stack already. The name does not convey what the purpose the base class serves, and we can't have a high level class in the lsst stack effectively reserving the name `Functor` for a specific purpose. Maybe even something as simple as `DataFrameFunctor`, though I guess that is not as descriptive as it could be.